src/modules/virtual_machines/linux_virtual_machine/tls_private_key.tf

```diff
@@ -1,5 +1,5 @@
 resource "tls_private_key" "main" {
-  for_each = var.settings.public_key_openssh
+  for_each = try(var.settings.public_key_openssh, {})
```

Comment:
Should we add a flag for login via password or via ssh keys which is different than `disable_password_authentication`?
With the highlighted change, you make public_key_openssh optional. Could that cause issues / misconfigurations?